src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
I think you're changing the semantics radically. In the previous version of this method, the ```form``` of the two bounds was completely orthogonal to the choice of denominator. They constituted bounds, upper and lower limits, and carried no further information. I think that was a good choice.